near-sdk/src/promise.rs

```diff
@@ -631,23 +631,44 @@ mod tests {
         "ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp".parse().unwrap()
     }
 
-    fn get_actions() -> std::vec::IntoIter<VmAction> {
+    fn get_actions() -> std::vec::IntoIter<MockAction> {
         let receipts = get_created_receipts();
         let first_receipt = receipts.into_iter().next().unwrap();
         first_receipt.actions.into_iter()
     }
+    fn convert_key(public_key: &PublicKey) -> near_crypto::PublicKey {
+        // TODO: add Secp conversion
+
+        let key_bytes = public_key.clone().into_bytes();
+        let public_key = near_crypto::PublicKey::ED25519(
+            near_crypto::ED25519PublicKey::try_from(&key_bytes.as_slice()[1..]).unwrap(),
+        );
+        public_key
+    }
 
     fn has_add_key_with_full_access(public_key: PublicKey, nonce: Option<u64>) -> bool {
         get_actions().any(|el| {
+            let public_key = convert_key(&public_key);
+
             matches!(
                 el,
-                VmAction::AddKeyWithFullAccess { public_key: p, nonce: n }
+                MockAction::AddKeyWithFullAccess { public_key: p, nonce: n, receipt_index: _, }
                 if p == public_key
                     && (nonce.is_none() || Some(n) == nonce)
             )
         })
     }
 
+    fn map_vec_str(vec_str: Vec<Vec<u8>>) -> Vec<String> {
+        vec_str
+            .into_iter()
+            .map(|element| {
+                let string: String = String::from_utf8(element).unwrap();
+                string
+            })
+            .collect()
+    }
+
```

Comment:
todo: move this to `From<near_vm_runner::logic::mocks::mock_external::MockAction>`